lib/fritzbox/smarthome/actor.rb

```diff
@@ -17,7 +17,7 @@ def all(types: ['group', 'device'])
           response = get(command: 'getdevicelistinfos')
           xml = nori.parse(response.body)
           Array.wrap(types.map { |type| xml.dig('devicelist', type) }.flatten).compact.map do |data|
-            klass = Actor.descendants.find { |k| k.match?(data) }
+            klass = Actor.descendants.find { |k| k.match?(data) } || self
```

Comment:
I've found one small issue with this change:

The current implementation makes sure to either return specific actor types only (when called from a specific sub-class) or all available ones (when called from the base-class).

This is why I have this weird `self.in?([klass, Actor])` construct in the next line.

Please just use `Actor` instead of `self` to fix that part.

```suggestion
            klass = Actor.descendants.find { |k| k.match?(data) } || Actor
```

Also to avoid introducing this regression in future we should add the unknown device also to `spec/support/fixtures/getheaterandgrouplistinfos.xml` which then would cover that case with the existing specs:

```diff
--- spec/support/fixtures/getheaterandgrouplistinfos.xml
+++ spec/support/fixtures/getheaterandgrouplistinfos.xml
@@ -71,4 +71,5 @@
       </nextchange>
     </hkr>
   </device>
+  <device identifier="12345 54321" id="4711" functionbitmask="007" fwversion="0815" productname="Sub-Etha Radio Transmitter"></device>
```

Reply:
@klausmeyer done in d3a12d2! :smile: